# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsAHttpTransaction.h
# Commit: 521c9e019678
# Full Hash: 521c9e019678f584590f63473416d40e03f33342
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 04:10:55
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r dfb39ca05529 -r 521c9e019678 netwerk/protocol/http/nsAHttpTransaction.h
--- a/netwerk/protocol/http/nsAHttpTransaction.h	Tue Mar 14 15:29:49 2023 +0200
+++ b/netwerk/protocol/http/nsAHttpTransaction.h	Tue Mar 14 12:59:40 2023 +0000
@@ -7,6 +7,8 @@
 
 #include "nsTArray.h"
 #include "nsWeakReference.h"
+#include "nsIRequest.h"
+#include "nsITRRSkipReason.h"
 
 #ifdef Status
 /* Xlib headers insist on this for some reason... Nuke it because
@@ -181,6 +183,8 @@
   virtual void ReuseConnectionOnRestartOK(bool) {}
   virtual void SetIsHttp2Websocket(bool) {}
   virtual bool IsHttp2Websocket() { return false; }
+  virtual void SetTRRInfo(nsIRequest::TRRMode aMode,
+                          TRRSkippedReason aSkipReason){};
 
   // We call this function if we want to use alt-svc host again on the next
   // restart. If this function is not called on the next restart the